bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -473,11 +473,10 @@ bool PreProcessor::checkClientMsgCorrectness(uint64_t reqSeqNum,
     return false;
   }
   const bool &invalidClient = !myReplica_.isValidClient(clientId);
-  const bool &sentFromReplicaToNonPrimary = myReplica_.isIdOfReplica(senderId) && !myReplica_.isCurrentPrimary();
```

Comment:
Currently, all the "internal requests" are not sent to preprocess (and we also don't yet have a mechanism for doing that).
Please add a comment in the description of the PR that we want to generalize the idea of letting replicas act like a client, but it is not relevant in preProcessor (for now)